undebt/pattern/interface.py

```diff
@@ -74,10 +76,35 @@ def patterns_from_files(pattern_files):
 
 def load_module(path):
     """Loads a module from its path."""
+    if module_like(path):
+        return _load_module(path)
+
     pattern_name = os.path.splitext(os.path.basename(path))[0]
     return imp.load_source(pattern_name, path)
 
 
+def _load_module(full_name):
+    try:
+        return importlib.import_module(full_name)
+    except ImportError as e:
+        raise e
+    finally:
+        # Unshim path
+        sys.path = sys.path[1:]
+
+
+# _module_re is not _strictly_ correct, but we do a quick check for strings
+# ending in .py before running the regex.
+_module_re = re.compile(r'^\w+(\.\w+)*$')
+
+
+def module_like(path):
+    if path.endswith('.py'):
+        return False
+
+    return bool(_module_re.match(path))
```

Comment:
I like this idea. We should be able to make this change without the UI changing, and it should make the module-loading part of the code simpler (now that I have made it more complicated)